File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/C++/s826491904_commented.cpp
#include <iostream>
#include <algorithm>

class Main {
public:
    void solve() {
        int N;
        std::cin >> N;
        int max, min;
        std::cin >> min; // Correctly read the first integer
        max = min;

        for (int i = 1; i < N; i++) {
            int in;
            std::cin >> in;
            max = std::max(max, in);
            min = std::min(min, in);
        }
        std::cout << max - min << std::endl;
    }
};

int main() {
    Main().solve();
    return 0; // Return 0 to indicate successful execution
    // End of Code
}

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Go/s826491904_commented.go
package main

import (
	"fmt"
)

func main() {
	var N int
	fmt.Scan(&N)
	var first int
	fmt.Scan(&first)
	max, min := first, first

	for i := 0; i < N-1; i++ {
		var num int
		fmt.Scan(&num)
		if num > max {
			max = num
		}
		if num < min {
			min = num
		}
	}
	fmt.Println(max - min)
}
// End of Code
